go/libraries/doltcore/sqle/dtables/column_diff_table.go

```diff
@@ -0,0 +1,649 @@
+		tableChange.dataChange,
+		tableChange.schemaChange,
+	), nil
+}
+
+// loadTableChanges loads the current commit's table changes and metadata
+// into the iterator.
+func (itr *doltColDiffCommitHistoryRowItr) loadTableChanges(ctx context.Context, commit *doltdb.Commit) error {
+	tableChanges, err := itr.calculateTableChanges(ctx, commit)
+	if err != nil {
+		return err
+	}
+
+	itr.tableChanges = tableChanges
+	itr.tableChangesIdx = 0
+	if len(tableChanges) == 0 {
+		return nil
+	}
+
+	meta, err := commit.GetCommitMeta(ctx)
+	if err != nil {
+		return err
+	}
+	itr.meta = meta
+
+	cmHash, err := commit.HashOf()
+	if err != nil {
+		return err
+	}
+	itr.hash = cmHash
+
+	return nil
+}
+
+// calculateTableChanges calculates the tables that changed in the specified commit, by comparing that
+// commit with its immediate ancestor commit.
+func (itr *doltColDiffCommitHistoryRowItr) calculateTableChanges(ctx context.Context, commit *doltdb.Commit) ([]tableColChange, error) {
+	if len(commit.DatasParents()) == 0 {
+		return nil, nil
+	}
+
+	toRootValue, err := commit.GetRootValue(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parent, err := itr.ddb.ResolveParent(ctx, commit, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	fromRootValue, err := parent.GetRootValue(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	deltas, err := diff.GetTableDeltas(ctx, fromRootValue, toRootValue)
+	if err != nil {
+		return nil, err
+	}
+
+	tableChanges := make([]tableColChange, len(deltas))
+	for i := 0; i < len(deltas); i++ {
+		change, err := processTableColDelta(itr.ctx, itr.ddb, deltas[i])
+		if err != nil {
+			return nil, err
+		}
+
+		tableChanges[i] = *change
+	}
+
+	// Not all commits mutate tables (e.g. empty commits)
+	if len(tableChanges) == 0 {
+		return nil, nil
+	}
+
+	return tableChanges, nil
+}
+
+// Close closes the iterator.
+func (itr *doltColDiffCommitHistoryRowItr) Close(*sql.Context) error {
+	return nil
+}
+
+// tableColChange is an internal data structure used to hold the results of processing
+// a diff.TableDelta structure into the output data for this system table.
+type tableColChange struct {
+	tableName    string
+	colNames     []string
+	dataChange   bool
+	schemaChange bool
+}
+
+// processTableColDelta processes the specified TableDelta to determine what kind of change it was (i.e. table drop,
+// table rename, table create, or data update) and returns a tableChange struct representing the change.
+func processTableColDelta(ctx *sql.Context, ddb *doltdb.DoltDB, delta diff.TableDelta) (*tableColChange, error) {
+	// Dropping a table is always a schema change, and also a data change if the table contained data
+	if delta.IsDrop() {
+		isEmpty, err := isTableDataEmpty(ctx, delta.FromTable)
+		if err != nil {
+			return nil, err
+		}
+
+		return &tableColChange{
+			tableName:    delta.FromName,
+			colNames:     delta.FromSch.GetAllCols().GetColumnNames(),
+			dataChange:   !isEmpty,
+			schemaChange: true,
+		}, nil
```

Comment:
small suggestion: drop, rename and add cases share very similar code with some repeated code, so this might be good part to be extracted out into its own function that all these cases can call.

Reply:
Along the same line... Taylor recently added some nice correctness fixes to `TableDelta`. You might be able to simplify this a little bit if you can reuse functions in `TableDelta`, such as `HasDataChange()` and `HasSchemaChange()`, but I haven't looked through the details enough to see if that'll fully work out.